Hi, the **Pre_test Static Analysis** pre-checker is failing on this PR due to a flake8 lint error in your change:

```
tests/vxlan/test_vxlan_vnet_bgp_subintf.py:535:102: E231 missing whitespace after ','
```

`/azp run` won't clear this since it's a real lint error in the code — please fix it (run `python3 -m flake8 <file> --max-line-length=120` locally to confirm clean), then push an update. Thanks!

--- a/tests/acl/test_src_mac_rewrite.py
+++ b/tests/acl/test_src_mac_rewrite.py
@@ -182,7 +182,8 @@ def fixture_setUp(rand_selected_dut, tbinfo, ptfadapter):
         tunnel_name=data['vxlan_tunnel_name'],
         src_ip=data['loopback_src_ip'],
         portchannel_name=selected_pc,
-        router_mac=rand_selected_dut.facts['router_mac']
+        router_mac=rand_selected_dut.facts['router_mac'],
+        asic_type=rand_selected_dut.facts.get('asic_type')
     )
 
     def _check_vnet_route(duthost):
@@ -474,7 +475,8 @@ def remove_acl_rules(duthost):
     pytest_assert(exists_output.strip() == "0", f"ACL rule {state_db_key} still exists in STATE_DB")
 
 
-def create_vxlan_vnet_config(duthost, tunnel_name, src_ip, portchannel_name="PortChannel101", router_mac=None):
+def create_vxlan_vnet_config(duthost, tunnel_name, src_ip, portchannel_name="PortChannel101", router_mac=None,
+                             asic_type=None):
     # --- VXLAN parameters ---
     vnet_base = VXLAN_VNI
     ptf_vtep = PTF_VTEP_IP
@@ -483,10 +485,16 @@ def create_vxlan_vnet_config(duthost, tunnel_name, src_ip, portchannel_name="Por
     ecmp_utils.Constants['KEEP_TEMP_FILES'] = True
     ecmp_utils.Constants['DEBUG'] = False
 
+    vxlan_tunnel_entry = {"src_ip": dut_vtep}
+    # On cisco-8000, base topology IP-in-IP decap tunnels may already use pipe TTL mode.
+    # Set VXLAN decap ttl_mode to pipe so orchagent passes DECAP_TTL_MODE consistently.
+    if asic_type == "cisco-8000":
+        vxlan_tunnel_entry["ttl_mode"] = "pipe"
+
     # --- Build overlay config JSON ---
     dut_json = {
         "VXLAN_TUNNEL": {
-            tunnel_name: {"src_ip": dut_vtep}
+            tunnel_name: vxlan_tunnel_entry
         },
         "VNET": {
             "Vnet1": {

--- a/tests/vxlan/test_scale_ecmp.py
+++ b/tests/vxlan/test_scale_ecmp.py
@@ -127,7 +127,12 @@ def vxlan_setup_one_vnet(duthost, ptfhost, tbinfo, cfg_facts,
 
     dut_vtep = get_loopback_ip(cfg_facts)
     logger.info(f"Creating VXLAN tunnel {TUNNEL_NAME} with source {dut_vtep}")
-    apply_chunk(duthost, {"VXLAN_TUNNEL": {TUNNEL_NAME: {"src_ip": dut_vtep}}}, "vxlan_tunnel")
+    vxlan_tunnel_entry = {"src_ip": dut_vtep}
+    # On cisco-8000, base topology IP-in-IP decap tunnels may already use pipe TTL mode.
+    # Set VXLAN decap ttl_mode to pipe so orchagent passes DECAP_TTL_MODE consistently.
+    if duthost.facts.get("asic_type") == "cisco-8000":
+        vxlan_tunnel_entry["ttl_mode"] = "pipe"
+    apply_chunk(duthost, {"VXLAN_TUNNEL": {TUNNEL_NAME: vxlan_tunnel_entry}}, "vxlan_tunnel")
     apply_chunk(duthost, {"VNET": {VNET_NAME: {"vni": str(VNI), "vxlan_tunnel": TUNNEL_NAME}}}, "vnet")
 
     ptf_port_index = port_indexes[ingress_if]

--- a/tests/vxlan/test_vnet_decap.py
+++ b/tests/vxlan/test_vnet_decap.py
@@ -106,7 +106,8 @@ def setup(request, duthosts, rand_one_dut_hostname, tbinfo, inner_ip_version, ou
     topo = tbinfo["topo"]["type"].upper()
     vnet_interface = ecmp_utils.select_required_interfaces(duthost, 1, minigraph_facts, outer_ip_version_str,
                                                            topo=topo)[0]
-    vxlan_tunnel = ecmp_utils.create_vxlan_tunnel(duthost, minigraph_facts, outer_ip_version_str)
+    vxlan_tunnel = ecmp_utils.create_vxlan_tunnel(duthost, minigraph_facts, outer_ip_version_str,
+                                                   ttl_mode="pipe" if asic_type == "cisco-8000" else None)
     inner_ip_version_str = f"v{inner_ip_version}"
     encap_type = f"{inner_ip_version_str}_in_{outer_ip_version_str}"
     vnet = next(iter(ecmp_utils.create_vnets(duthost, vxlan_tunnel,

--- a/tests/vxlan/test_vnet_decap.py
+++ b/tests/vxlan/test_vnet_decap.py
@@ -107,7 +107,7 @@ def setup(request, duthosts, rand_one_dut_hostname, tbinfo, inner_ip_version, ou
     vnet_interface = ecmp_utils.select_required_interfaces(duthost, 1, minigraph_facts, outer_ip_version_str,
                                                            topo=topo)[0]
     vxlan_tunnel = ecmp_utils.create_vxlan_tunnel(duthost, minigraph_facts, outer_ip_version_str,
-                                                   ttl_mode="pipe" if asic_type == "cisco-8000" else None)
+                                                  ttl_mode="pipe" if asic_type == "cisco-8000" else None)
     inner_ip_version_str = f"v{inner_ip_version}"
     encap_type = f"{inner_ip_version_str}_in_{outer_ip_version_str}"
     vnet = next(iter(ecmp_utils.create_vnets(duthost, vxlan_tunnel,

--- a/tests/vxlan/test_vxlan_bfd_tsa.py
+++ b/tests/vxlan/test_vxlan_bfd_tsa.py
@@ -165,7 +165,8 @@ def fixture_setUp(duthosts,
         tunnel_names[outer_layer_version] = ecmp_utils.create_vxlan_tunnel(
             data['duthost'],
             minigraph_data=minigraph_facts,
-            af=outer_layer_version)
+            af=outer_layer_version,
+            ttl_mode="pipe" if data['duthost'].facts.get("asic_type") == "cisco-8000" else None)
 
     payload_version = ecmp_utils.get_payload_version(encap_type)
     encap_type = "{}_in_{}".format(payload_version, outer_layer_version)

--- a/tests/vxlan/test_vxlan_ecmp.py
+++ b/tests/vxlan/test_vxlan_ecmp.py
@@ -252,7 +252,8 @@ def fixture_setUp(duthosts,
         tunnel_names[outer_layer_version] = ecmp_utils.create_vxlan_tunnel(
             data['duthost'],
             minigraph_data=minigraph_facts,
-            af=outer_layer_version)
+            af=outer_layer_version,
+            ttl_mode="pipe" if data['duthost'].facts.get("asic_type") == "cisco-8000" else None)
 
     payload_version = ecmp_utils.get_payload_version(encap_type)
     encap_type = "{}_in_{}".format(payload_version, outer_layer_version)

--- a/tests/vxlan/test_vxlan_ecmp_switchover.py
+++ b/tests/vxlan/test_vxlan_ecmp_switchover.py
@@ -134,7 +134,8 @@ def fixture_setUp(duthosts,
         tunnel_names[outer_layer_version] = ecmp_utils.create_vxlan_tunnel(
             data['duthost'],
             minigraph_data=minigraph_facts,
-            af=outer_layer_version)
+            af=outer_layer_version,
+            ttl_mode="pipe" if data['duthost'].facts.get("asic_type") == "cisco-8000" else None)
 
     payload_version = ecmp_utils.get_payload_version(encap_type)
     encap_type = "{}_in_{}".format(payload_version, outer_layer_version)

--- a/tests/vxlan/test_vxlan_route_advertisement.py
+++ b/tests/vxlan/test_vxlan_route_advertisement.py
@@ -131,7 +131,8 @@ def fixture_setUp(duthosts,
         tunnel_names[outer_layer_version] = ecmp_utils.create_vxlan_tunnel(
             data['duthost'],
             minigraph_data=minigraph_facts,
-            af=outer_layer_version)
+            af=outer_layer_version,
+            ttl_mode="pipe" if data['duthost'].facts.get("asic_type") == "cisco-8000" else None)
 
     payload_version = ecmp_utils.get_payload_version(encap_type)
     encap_type = "{}_in_{}".format(payload_version, outer_layer_version)

--- a/tests/vxlan/test_vxlan_tunnel_route_scale.py
+++ b/tests/vxlan/test_vxlan_tunnel_route_scale.py
@@ -218,7 +218,12 @@ def vxlan_setup_config(config_facts, cfg_facts, duthost, dut_indx, ptfhost,
     dut_vtep = get_loopback_ip(cfg_facts)
     ptf_vtep = PTF_VTEP
     vxlan_tun = TUNNEL_NAME
-    apply_chunk(duthost, {"VXLAN_TUNNEL": {vxlan_tun: {"src_ip": dut_vtep}}}, "vxlan_tunnel")
+    vxlan_tunnel_entry = {"src_ip": dut_vtep}
+    # On cisco-8000, base topology IP-in-IP decap tunnels may already use pipe TTL mode.
+    # Set VXLAN decap ttl_mode to pipe so orchagent passes DECAP_TTL_MODE consistently.
+    if duthost.facts.get("asic_type") == "cisco-8000":
+        vxlan_tunnel_entry["ttl_mode"] = "pipe"
+    apply_chunk(duthost, {"VXLAN_TUNNEL": {vxlan_tun: vxlan_tunnel_entry}}, "vxlan_tunnel")
 
     vnet_ptf_map = {}
 

--- a/tests/vxlan/test_vxlan_vnet_bgp_subintf.py
+++ b/tests/vxlan/test_vxlan_vnet_bgp_subintf.py
@@ -619,10 +619,13 @@ def setup_vnets(duthost, ptfhost, num_vnets, tunnel, base_vni):
 
 
 def setup_vxlan_tunnel(duthost, ptfhost, name, src_ip):
+    tunnel_entry = {"src_ip": src_ip}
+    # On cisco-8000, base topology IP-in-IP decap tunnels may already use pipe TTL mode.
+    # Set VXLAN decap ttl_mode to pipe so orchagent passes DECAP_TTL_MODE consistently.
+    if duthost.facts.get("asic_type") == "cisco-8000":
+        tunnel_entry["ttl_mode"] = "pipe"
     gnmi_set_update_config_db_json(duthost, ptfhost, f"{GNMI_PATH_PREFIX}/VXLAN_TUNNEL", {
-        name: {
-            "src_ip": src_ip
-        }
+        name: tunnel_entry
     }, "vxlan")
 
 

--- a/tests/vxlan/test_vxlan_vnet_bgp_subintf.py
+++ b/tests/vxlan/test_vxlan_vnet_bgp_subintf.py
@@ -532,7 +532,7 @@ def setup_portchannel_subintfs(duthost, ptfhost, portchannel_info, vnet_vnis, ba
                 gnmi_set_update_config_db_json(
                     duthost,
                     ptfhost,
-                    f"{GNMI_PATH_PREFIX}/VLAN_SUB_INTERFACE/{subintf_name}|{dut_ips[j][i].replace('/','~1')}",
+                    f"{GNMI_PATH_PREFIX}/VLAN_SUB_INTERFACE/{subintf_name}|{dut_ips[j][i].replace('/', '~1')}",
                     {},
                     f"{subintf_name}_ip")
 

